### PR TITLE
SPLAT-858 Provider cert tool results push

### PR DESCRIPTION
**Pull request by bostrt**
The OpenShift Provider Certification tool is run periodically in OpenShift CI. The certification results that are captured in each periodic job need to be saved long-term (and easily searchable/filterable) for engineers certifying OpenShift on new infrastructure providers. The certification results also need to be saved long-term for historical reference on quality (of both OpenShift and Certification program). 

This PR adds changes that will upload the archive to an S3 bucket.

### CODE REVIEW

File: ci-operator/step-registry/provider-certification-tool/save/provider-certification-tool-save-commands.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+RESULTS=$(ls "${ARTIFACT_DIR}/certification-results/*.tar.gz")
+RESULTS_URL="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/${JOB}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/provider-certification-tool-run/artifacts/certification-results/${RESULTS}"
+TOKEN=$(cat /var/run/vault/opct/github-token)
+VERSION=$(oc get clusterversion version -o=jsonpath='{.status.desired.version}')
+
+curl -X POST \
+-H "Accept: Accept: application/vnd.github+json" \
+-H "Authorization: Bearer ${TOKEN}" \
+https://api.github.com/repos/redhat-openshift-ecosystem/provider-certification-tool/actions/workflows/push-image.yaml/dispatches \
+-d '{"ref":"push-results","inputs":{"results_url":"'"${RESULTS_URL}"'","cluster_version":"'"${VERSION}"'","platformType":"'"${CLUSTER_TYPE}"'"}'
```
Comment (bostrt) [inline ci-operator/step-registry/provider-certification-tool/save/provider-certification-tool-save-commands.sh:18]: Update `ref` after GitHub Action PR merges.

File: ci-operator/step-registry/provider-certification-tool/save/provider-certification-tool-save-commands.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
```
Comment (bostrt) [inline ci-operator/step-registry/provider-certification-tool/save/provider-certification-tool-save-commands.sh:6]: Add condition to only execute for periodic jobs

**Comment by bostrt** [comment]
The GH Action was not triggered :facepalm:

**Comment by bostrt** [comment]
again! 
/test pj-rehearse

**Comment by bostrt** [comment]
It looks like the last test was successful except for the S3 bucket being pruned. A ticket has been filed requesting to preserve the bucket. I'll re-run the test now since I've manually recreated the S3 bucket folder. 

/test pj-rehearse

**Comment by bostrt** [comment]
@mtulio mind taking a look at this now that I'm using S3 instead of quay?

**Comment by bostrt** [comment]
AWS resource preservation has been enabled. Running test again to confirm its all setup okay:
/test pj-rehearse

File: ci-operator/step-registry/provider-certification-tool/run/provider-certification-tool-run-commands.sh [hunk 3]
```diff
@@ -20,3 +20,50 @@ mkdir -p "${ARTIFACT_DIR}/certification-results"
...
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+VERSION=$(oc get clusterversion version -o=jsonpath='{.status.desired.version}')
+
+# upload to AWS S3
+DATE=$(date +%Y%m%d)
+
+echo "s3://openshift-provider-certification/baseline-results/${VERSION}-${DATE}.tar.gz"
+echo '{"platform-type":"'"${CLUSTER_TYPE}"'"}'
+
+aws s3 cp "${ARTIFACT_DIR}"/certification-results/*.tar.gz "s3://openshift-provider-certification/baseline-results/${VERSION}-${DATE}.tar.gz" \
```
Comment (mtulio) [inline ci-operator/step-registry/provider-certification-tool/run/provider-certification-tool-run-commands.sh:68]: I am still uncertain if we need to add the baseline platform on the name of the results - as it is external, I can, for example, run on AWS and upload to the bucket the results. But metadata for now sgtm.

**Review by mtulio — COMMENTED** [review]
Let's update the credentials to the dedicated user, and check the next execution.
Overall, LGTM

File: ci-operator/step-registry/provider-certification-tool/run/provider-certification-tool-run-commands.sh (hunk 3 above)
Comment (bostrt) [reply ci-operator/step-registry/provider-certification-tool/run/provider-certification-tool-run-commands.sh:68]: Sounds good to me as well. I think as long as we don't heavily depend on the metadata somehow we can change that up later as requirements change.

**Comment by bostrt** [comment]
Credentials have been updated, testing once more
 /test pj-rehearse

**Comment by mtulio** [comment]
/lgtm
Screenshot from 2022-11-18 13-09-53

**Pull request merged**